src/darker/highlighting/__init__.py

```diff
@@ -1,12 +1,54 @@
 """Highlighting of terminal output"""
 
-try:
-    import pygments  # noqa: F401
-except ImportError:
-    from darker.highlighting import without_pygments
+# pylint: disable=import-outside-toplevel,unused-import
 
-    colorize = without_pygments.colorize
-else:
-    from darker.highlighting import with_pygments
+import sys
+from typing import Optional, cast
 
-    colorize = with_pygments.colorize
+
+def should_use_color(config_color: Optional[bool]) -> bool:
+    """Return ``True`` if configuration and package support allow output highlighting
+
+    In ``config_color``, the combination of ``color =`` in ``pyproject.toml``, the
+    ``PY_COLORS`` environment variable, and the ``--color``/``--no-color`` command line
+    options determine whether the user wants to force enable or disable highlighting.
+
+    If highlighting isn't forced either way, it is automatically enabled for terminal
+    output.
+
+    Finally, if ``pygments`` isn't installed, highlighting is disabled.
+
+    :param config_color: The configuration as parsed from ``pyproject.toml`` and
+                         overridden using environment variables and/or command line
+                         options
+    :return: ``True`` if highlighting should be used
+
+    """
+    if config_color is not None:
+        use_color = config_color
+    else:
+        use_color = sys.stdout.isatty()
+    if use_color:
+        try:
+            import pygments  # noqa
+
+            return True
+        except ImportError:
+            pass
+    return False
```

Comment:
```suggestion
        except ImportError:
            return False
```

Reply:
Darker will only support `PY_COLORS=0` and `PY_COLORS=1`, see [discussion here]. Thanks @MatthijsBurgh for the suggestion!

Reply:
I see that my suggestion has a different behaviour. I missed the last `if`.